server/server.go

```diff
@@ -26,9 +26,12 @@ func Run(appContext *system.AppContext) {
 
 	drawRoutes(r, appContext)
 
+	h := util.NewCorsHandler(r, appContext.Config.Cors.AllowedOrigins)
+	h = util.NewMethodOverrider(h)
```

Comment:
I think we should override a method before the cors handler in order to determine if the request that has overridden is valid.